### ISSUE TITLE
Manually editing XML when adding Storage fails

### ISSUE THREAD

**Issue by BarryStokes** [virt-manager/virt-manager#157]
I was adding a new storage device to a remotely connected VM.

Steps:
1) Go to Hardware Details of VM
2) Click "Add Hardware"
3) In the "Add New Virtual Hardware" window, ensure Storage is selected
4) Click "Select or create custom storage"
5) Click "Manage..." to browse available images
6) Select required image and click "Choose Volume"
7) Change "Device type" to "Floppy device"
8) Click "XML" to change to XML view
9) Click "Finish"

The system then presents the following error: "Error validating device parameters: Must specify storage creation parameters for non-existent path '<REDACTED>'."

Traceback:
"Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/addhardware.py", line 1382, in _finish
    if self._validate_device(dev) is False:
  File "/usr/share/virt-manager/virtManager/addhardware.py", line 1422, in _validate_device
    dev.validate()
  File "/usr/share/virt-manager/virtinst/devices/disk.py", line 751, in validate
    raise ValueError(
ValueError: Must specify storage creation parameters for non-existent path '<REDACTED>'."

Expected outcome:
The device would be added.

If you switch back from XML view to Details view and click Finish the operation succeeds, but leaves you unable to make XML changes at the point of adding the storage.

Once the device has been added it is possible to manually amend the XML, so it looks like there are different validations taking place.

I have only tested this for adding a floppy device to the system.

**Comment by crobinso** [comment]
Thanks for the report, I can't reproduce though. Are you using virt-manager 3.0.0? You can test git with

```
git clone https://github.com/virt-manager/virt-manager
cd virt-manager
./virt-manager --debug
```

If it still reproduces that way, please provide:
* The starting XML in the editor, before you make any changes
* The XML after you make changes, where pressing 'Finish' generates the error

**Comment by BarryStokes** [comment]
My apologies, I was just using the latest version shipped with my OS' package manager, which is 2.2.1.

Just tried it with the latest git version and can't reproduce the error I was seeing.
